core/commonMain/src/kotlinx/serialization/internal/BuiltInSerializers.kt

```diff
@@ -23,3 +24,16 @@ internal object DurationSerializer : KSerializer<Duration> {
         return Duration.parseIsoString(decoder.decodeString())
     }
 }
+
+@PublishedApi
+internal object NothingSerializer : KSerializer<Nothing> {
+    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("kotlin.Nothing", PrimitiveKind.INT)
```

Comment:
Have we discussed this design? I'm not sure reusing PrimitiveKind.INT is good idea

Reply:
New primitive kind?

Reply:
#2150